### PR TITLE
add EventsLogger option in loggedCluster package

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
### What this PR does / why we need it

Towards https://github.com/giantswarm/roadmap/issues/3750

Split the [huge PR](https://github.com/giantswarm/logging-operator/pull/263) into smaller ones.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Follow deployment test procedure in the tests/manual_e2e directory and have a working branch.

### CODE REVIEW

**Comment by QuentinBisson** [comment]
Can you fix the tests?

**Comment by QuantumEnigmaa** [comment]
> Can you fix the tests?

Yeah I'm on it :) 

Build failing because of the following : 
```
internal/controller/vintagewc_controller.go:79:44: cannot use loggedCluster (variable of type *vintagewc.Object) as loggedcluster.Interface value in argument to r.LoggingReconciler.Reconcile: *vintagewc.Object does not implement loggedcluster.Interface (missing method GetEventsLogger)
```

**Comment by QuantumEnigmaa** [comment]
@QuentinBisson Problem fixed

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - Add support for Private CAs in alloy logs.
+- Add EventsLogger option and related methods in loggedCLuster package.
```
Comment (QuentinBisson) [inline CHANGELOG.md:13]:
```suggestion
- Add KubeEventsLogger option and related methods in loggedCLuster package.
```

**Review by QuentinBisson — APPROVED** [review]
LGTM

**Review by hervenicol — APPROVED** [review]
Fine with me :)

**Pull request merged**